pkg/asset/installconfig/aws/route53.go

```diff
@@ -148,37 +157,227 @@ func GetR53ClientCfg(sess *awss.Session, roleARN string) *aws.Config {
+			EvaluateTargetHealth: aws.Bool(false),
+		})
+	}
+	input := &route53.ChangeResourceRecordSetsInput{
+		HostedZoneId: aws.String(zoneID),
 		ChangeBatch: &route53.ChangeBatch{
-			Comment: aws.String(fmt.Sprintf("Creating record for api and api-int in domain %s", ic.ClusterDomain())),
-		},
-		HostedZoneId: zone.Id,
-	}
-	for _, prefix := range []string{"api", "api-int"} {
-		params.ChangeBatch.Changes = append(params.ChangeBatch.Changes, &route53.Change{
-			Action: aws.String("UPSERT"),
-			ResourceRecordSet: &route53.ResourceRecordSet{
-				Name: aws.String(fmt.Sprintf("%s.%s.", prefix, ic.ClusterDomain())),
-				Type: aws.String("A"),
-				AliasTarget: &route53.AliasTarget{
-					DNSName:              aws.String(target),
-					HostedZoneId:         aws.String(hostedZoneIDPerRegionNLBMap[ic.AWS.Region]),
-					EvaluateTargetHealth: aws.Bool(true),
+			Comment: aws.String(fmt.Sprintf("Creating record %s", name)),
+			Changes: []*route53.Change{
+				{
+					Action:            aws.String("UPSERT"),
+					ResourceRecordSet: recordSet,
 				},
 			},
-		})
+		},
 	}
+	res, err := client.ChangeResourceRecordSetsWithContext(ctx, input)
+	if err != nil {
+		return nil, err
+	}
+
+	return res.ChangeInfo, nil
+}
+
+// HostedZoneInput defines the input parameters for hosted zone creation.
+type HostedZoneInput struct {
+	Name     string
+	InfraID  string
+	VpcID    string
+	Region   string
+	Role     string
+	UserTags map[string]string
+}
+
+// CreateHostedZone creates a private hosted zone.
+func (c *Client) CreateHostedZone(ctx context.Context, input *HostedZoneInput) (*route53.HostedZone, error) {
+	cfg := GetR53ClientCfg(c.ssn, input.Role)
 	svc := route53.New(c.ssn, cfg)
-	if _, err := svc.ChangeResourceRecordSets(params); err != nil {
-		return fmt.Errorf("failed to create records for api/api-int: %w", err)
+
+	callRef := fmt.Sprintf("%d", time.Now().Unix())
+	res, err := svc.CreateHostedZoneWithContext(ctx, &route53.CreateHostedZoneInput{
+		CallerReference: aws.String(callRef),
+		Name:            aws.String(input.Name),
+		HostedZoneConfig: &route53.HostedZoneConfig{
+			PrivateZone: aws.Bool(true),
+			Comment:     aws.String("Created by Openshift Installer"),
+		},
+		VPC: &route53.VPC{
+			VPCId:     aws.String(input.VpcID),
+			VPCRegion: aws.String(input.Region),
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error creating private hosted zone: %w", err)
 	}
-	return nil
+
+	if res == nil {
+		return nil, fmt.Errorf("error creating private hosted zone: %w", err)
+	}
+
+	// Tag the hosted zone
+	tags := mergeTags(input.UserTags, map[string]string{
+		"Name": fmt.Sprintf("%s-int", input.InfraID),
+	})
+	_, err = svc.ChangeTagsForResourceWithContext(ctx, &route53.ChangeTagsForResourceInput{
+		ResourceType: aws.String("hostedzone"),
+		ResourceId:   res.HostedZone.Id,
+		AddTags:      r53Tags(tags),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("failed to tag private hosted zone: %w", err)
+	}
+
+	// Set SOA minimum TTL
```

Comment:
The SOA TTL modification seems to be very specific thing. Is it make sense to reuse some pieces from[1]  `pkg/infrastructure/aws/dns.go` ?

[1] https://github.com/openshift/installer/blob/39ba56297d995b1f3e51ef86f5e7c8aa4436316c/pkg/infrastructure/aws/dns.go#L121-L150

Reply:
I thought about that, but that code was added for the AWS SDK provisioning. Once capi/aws is in place, we should get rid of the former. So I decided to just duplicate the code for now.